packages/task/src/browser/quick-open-task.ts

```diff
@@ -468,87 +378,75 @@ export class QuickOpenTask implements QuickOpenModel, QuickOpenHandler {
     }
 }
 
-export class TaskRunQuickOpenItem extends QuickOpenGroupItem {
-
+export class TaskRunQuickOpenItem implements monaco.quickInput.IAnythingQuickPickItem {
     constructor(
-        protected readonly token: number,
-        protected readonly task: TaskConfiguration,
+        public readonly token: number,
+        public readonly task: TaskConfiguration,
         protected taskService: TaskService,
         protected isMulti: boolean,
-        public readonly options: QuickOpenGroupItemOptions,
         protected readonly taskDefinitionRegistry: TaskDefinitionRegistry,
         protected readonly taskNameResolver: TaskNameResolver,
-        protected readonly taskSourceResolver: TaskSourceResolver
-    ) {
-        super(options);
-    }
-
-    getTask(): TaskConfiguration {
-        return this.task;
-    }
+        protected readonly taskSourceResolver: TaskSourceResolver,
+        protected taskConfigurationManager: TaskConfigurationManager,
+        public readonly buttons?: Array<QuickInputButton>
+    ) { }
 
-    getLabel(): string {
+    get label(): string {
         return this.taskNameResolver.resolve(this.task);
     }
 
-    getToken(): number {
-        return this.token;
+    get description(): string {
+        return renderScope(this.task._scope, this.isMulti);
     }
 
-    getGroupLabel(): string {
-        return this.options.groupLabel || '';
+    get detail(): string | undefined {
+        return this.task.detail;
     }
 
-    getDescription(): string {
-        return this.options.description || renderScope(this.task._scope, this.isMulti);
+    accept(): void {
+        this.execute();
     }
 
-    run(mode: QuickOpenMode): boolean {
-        if (mode !== QuickOpenMode.OPEN) {
-            return false;
-        }
-
+    execute(): void {
         const scope = this.task._scope;
         if (this.taskDefinitionRegistry && !!this.taskDefinitionRegistry.getDefinition(this.task)) {
             this.taskService.run(this.token, this.task.source || this.task._source, this.task.label, scope);
         } else {
             this.taskService.run(this.token, this.task._source, this.task.label, scope);
         }
-        return true;
     }
 
-    getDetail(): string | undefined {
-        return this.task.detail;
+    trigger(): monaco.quickInput.TriggerAction {
+        this.taskService.configure(this.token, this.task);
```

Comment:
what use case this method should handle? 
is it `Configure task` action when the `gear` icon is clicked?

I'm asking because I still see some problems for the `gear` icon for:
- `Run Build Task...`
- `Run Test Task...`
- `Run Last Task`

The `gear` icon is displayed for items, but nothing happens when a user clicks on the icon... 
![gear_icon](https://user-images.githubusercontent.com/5676062/122984294-84711480-d3a5-11eb-9ef9-ded22bafa84a.gif)


For the `master` branch I see that the icon is NOT displayed for: the `Run Build Task...` and  `Run Test Task...` actions. It's displayed for the `Run Last Task` action and has a handler which runs `Configure Task` action.

Reply:
> what use case this method should handle?
> is it `Configure task` action when the `gear` icon is clicked?

the `trigger` method handle the `gear` icon click event only when the quick list is opened by `this.quickInputService?.open(...)` but when using `this.quickInputService?.showQuickPick(...)` we handle the `gear` event by giving the handler in the method options `onDidTriggerItemButton` like done [here] (https://github.com/danarad05/theia/blob/9580fefb4c5460fb97a8760d9b3be1c0c7d91f39/packages/task/src/browser/quick-open-task.ts#L112-L119), Both of them execute the `Configure task` action...
> I'm asking because I still see some problems for the `gear` icon for:
> 
> * `Run Build Task...`
> * `Run Test Task...`
> * `Run Last Task`
> 
> The `gear` icon is displayed for items, but nothing happens when a user clicks on the icon...

I'll add the `onDidTriggerItemButton` handler there too..
> 
> For the `master` branch I see that the icon is NOT displayed for: the `Run Build Task...` and `Run Test Task...` actions. It's displayed for the `Run Last Task` action and has a handler which runs `Configure Task` action.
`Run Build Task` in VsCode does have the gear icon (this `monaco` version supports the configuration of the `gear` icon.. which wasn't before)

Reply:
@EstherPerelman since we're using a `PickerQuickAccessProvider", we can return `IPickerQuickAccessItem`s from the "getItems()" method, which has a method for handling clicks on the icon. That way, we're guaranteed to always handle those clicks. I guess we could then remove the `showRunTask` and just rely on QuickAccessController.open(prefix).

Reply:
After going over this with @EstherPerelman :
1. `showRunTask()` method will be removed and we will use  `open()` (in quick-open-task.ts) - as you are right and it is redundant 
2. Concerning your remark about `IPickerQuickAccessItems`, from what I understand there are 2 ways of opening quickInput widget (which I implemented in similar way as is in VS Code):
a) use `quickAccess.show(PREFIX)` which when invoked is using `IPickerQuickAccessItem` and with it invokes the "trigger/accept" methods.
b) use  'showQuickPick" method (monaco-quick-input-service.ts) which creates a QuickPick (using createQuickPick() method) which isn't using the `IPickerQuickAccessItem` but the `IQuickPickItem` and therefore isn't using the `trigger/accept` methods but the `onDidTriggerItemButton` handler. 

I personally would prefer Theia to implement this functionality similarly to VSCode' and not "break compatibility" with it. Passing a `IPickerQuickAccessItem` to `showQuickPick` method would need obviously to change implementation so that `trigger/accept` methods would be invoked (as they won't be invoked automatically) --- not that this is difficult - but again it diverts Theia's implementation to a more proprietary one and less similar to VSCode's.

Reply:
Since `IPickerQuickAccessItems` extends the `QuickPickItem` interface we can safely use it wherever we use `QuickPickItem` now. I don't understand how that would be less compatible with VS Code. What is your concern concretely? When we use "showQuickPick", we can pass in the `onDidTriggerItemButton` option and simply check whether the item has a "trigger" method. 
If we want to have the "configure" functionality for all quick picks with tasks, we should route the "showQuickPick" through a shared method that sets up the button trigger. Currently there are eight invocation sites in quick-open-task.ts.

Reply:
But this is not a hill I'm going to die on. If we can fix the functionality, I'm fine with that.

Reply:
> Since IPickerQuickAccessItems extends the QuickPickItem interface we can safely use it wherever we use QuickPickItem now

That is of course correct - we can use IPickerQuickAccessItems or any interface we like but my point is that it is deviating from original way of using these apis in VSCode. For instance - in previous version, Theia had an entirely specific implementation of `quickOpen` and upgrading Theia to `quickInput` was a substantial effort - from this point of view - I am hesitant about deviating from original implementation as later changes in VSCode might not be implemented in Theia as easily. If we are basing Theia on VSCode - then IMO we should implement same interfaces/apis - as closely as possible.

> Currently there are eight invocation sites in quick-open-task.ts.

Yes, I agree that for sure quick-open-task.ts and other files could be improved more - but I don't think this is the PR to do so - it is large enough as it is. we need to complete this PR and afterwards do improvements. That is of course just my opinion.

Reply:
> But this is not a hill I'm going to die on. If we can fix the functionality, I'm fine with that.

@tsmaeder Meantime I pushed the fix, I will revert if needed...